src/js/msp.js

```diff
@@ -27,112 +60,166 @@ var MSP = {
+                this.message_length_received++;
+
+                if (this.message_length_received >= this.message_length_expected) {
+                    this.state = this.state === this.decoder_states.PAYLOAD_V1 ?
+                        this.decoder_states.CHECKSUM_V1 :
+                        this.decoder_states.CHECKSUM_V2;
+                }
+                break;
+            case this.decoder_states.CHECKSUM_V1:
+                if (this.message_length_expected >= this.constants.JUMBO_FRAME_MIN_SIZE) {
+                    this.message_checksum = this.constants.JUMBO_FRAME_MIN_SIZE;
+                } else {
+                    this.message_checksum = this.message_length_expected;
+                }
+                this.message_checksum ^= this.code;
+                if (this.message_length_expected >= this.constants.JUMBO_FRAME_MIN_SIZE) {
+                    this.message_checksum ^= this.message_length_expected & 0xFF;
+                    this.message_checksum ^= (this.message_length_expected & 0xFF00) >> 8;
+                }
+                for (let ii = 0; ii < this.message_length_received; ii++) {
+                    this.message_checksum ^= this.message_buffer_uint8_view[ii];
+                }
+                this._dispatch_message(data[i]);
+                break;
+            case this.decoder_states.CHECKSUM_V2:
+                this.message_checksum = 0;
+                this.message_checksum = this.crc8_dvb_s2(this.message_checksum, 0); // flag
+                this.message_checksum = this.crc8_dvb_s2(this.message_checksum, this.code & 0xFF);
+                this.message_checksum = this.crc8_dvb_s2(this.message_checksum, (this.code & 0xFF00) >> 8);
+                this.message_checksum = this.crc8_dvb_s2(this.message_checksum, this.message_length_expected & 0xFF);
+                this.message_checksum = this.crc8_dvb_s2(this.message_checksum, (this.message_length_expected & 0xFF00) >> 8);
+                for (let ii = 0; ii < this.message_length_received; ii++) {
+                    this.message_checksum = this.crc8_dvb_s2(this.message_checksum, this.message_buffer_uint8_view[ii]);
+                }
+                this._dispatch_message(data[i]);
+                break;
+            default:
+                console.log('Unknown state detected: ' + this.state);
             }
         }
         this.last_received_timestamp = Date.now();
     },
+    _initialize_read_buffer: function() {
+        this.message_buffer = new ArrayBuffer(this.message_length_expected);
+        this.message_buffer_uint8_view = new Uint8Array(this.message_buffer);
+    },
+    _dispatch_message: function(expectedChecksum) {
+        if (this.message_checksum === expectedChecksum) {
+            // message received, store dataview
+            this.dataView = new DataView(this.message_buffer, 0, this.message_length_expected);
+        } else {
+            console.log('code: ' + this.code + ' - crc failed');
```

Comment:
Strange that Sonar didn't detect this string concatenation...

Reply:
I've just changed all `console.log()` calls in `msp.js` to use string interpolation.